client/landing/stepper/declarative-flow/internals/steps-repository/components/migration-survey/index.tsx

```diff
@@ -11,17 +13,22 @@ const linkByCountry = {
 
 type Countries = keyof typeof linkByCountry;
 
-type MigrationSurveyProps = Pick< SurveyProps, 'isOpen' > & {
-	countryCode: string;
-};
+type MigrationSurveyProps = Pick< SurveyProps, 'isOpen' >;
 
 const getLink = ( country: Countries ) => {
 	return linkByCountry[ country ];
 };
 
-const MigrationSurvey = ( { isOpen, countryCode }: MigrationSurveyProps ) => {
+const MigrationSurvey = ( { isOpen }: MigrationSurveyProps ) => {
+	const isEnLocale = useIsEnglishLocale();
+	const { data } = useGeoLocationQuery();
+	const countryCode = data?.country_short;
```

Comment:
Here it would be elegant to use a `select`-or on the query:
```js
const { data: countryCode } = useGeoLocationQuery( {
  select: ( data ) => data.country_short
} );
```

Reply:
> Here it would be elegant to use a `select`-or on the query:

I like this idea as well, but since we [initialize `useQuery` with `GeoLocationData`](https://github.com/Automattic/wp-calypso/blob/cfe4f38279b7984d82a227ee1d4fd7e4afc16184/client/data/geo/use-geolocation-query.ts#L13) as with a generic type `GeoLocationData`, TypeScript protests when the return value of `select` doesn't satisfy the type.

```
Type '(data: GeoLocationData) => string' is not assignable to type '(data: GeoLocationData) => GeoLocationData'.
  Type 'string' is not assignable to type 'GeoLocationData'.ts(2322)
```